clang/test/CodeGenOpenCL/sampled_image.cl

```diff
@@ -1,5 +1,5 @@
-// RUN: %clang_cc1 %s -triple x86_64-unknown-linux-gnu -O0 -emit-llvm -o - | FileCheck %s
-// RUN: %clang_cc1 %s -triple x86_64-unknown-linux-gnu -O0 -emit-llvm -o - -cl-std=clc++ | FileCheck %s
+// RUN: %clang_cc1 %s -triple x86_64-unknown-linux-gnu -no-opaque-pointers -O0 -emit-llvm -o - | FileCheck %s
```

Comment:
What's the value of running this test with '-no-opaque-pointers' in builds where opaque pointers are enabled? Why not just mark it as unsupported in that case?

Reply:
I did this to follow the approach to handle the rest of OpenCL tests, which are not switched to opaque pointers yet.
Marking unsupported won't work because this test invokes the front-end compiler. We don't adjust front-end default for opaque pointers support (see clang/include/clang/Driver/Options.td), so we can change it only with explicit option. As you can see from the test results just a few tests requires adjustment and the rest of them are already updated to work with ON default value for opaque pointers support in FE. Unsupported mark works for the test invoking the driver.

It looked more closely at the tests and realized that CodeGenSPIRV tests are outdated, so I removed them (see commit messages for https://github.com/intel/llvm/pull/6378/commits/a926b481e458fe4dea16d943fceef74965fde63c and https://github.com/intel/llvm/pull/6378/commits/b41c3c70d0912cced066c653b726fe6626251a4d for more details). I updated the checks in the rest of the tests to handle opaque pointers (including the OpenCL test we added).